### PR TITLE
add nice approximation of the shippable/testrunner pep8 spec

### PR DESCRIPTION
**Pull request by tedder**
##### ISSUE TYPE
 - Docs Pull Request

##### COMPONENT NAME
testing/shippable/pep8

##### ANSIBLE VERSION
2.3.0 devel

##### SUMMARY
I needed to short-circuit the pep8 testing so I could fix things and iterate. Passing on, because The More You Know.

### CODE REVIEW

**Comment by sivel** [comment]
The test runner can be run directly to perform the pep8 test 

```
./test/runner/ansible-test sanity -v --test pep8
```

That would be the preferred command to execute as it is sure to be in sync with the way we are running pep8.

**Comment by tedder** [comment]
Perfect. I've updated it, @sivel. I find these type of breadcrumbs are handy for those of us who aren't in the code every day.

**Comment by abadger** [comment]
@mattclay I know you were working on test documentation but if that's a ways off maybe we want this as a short-term hint until the new docs can be merged?

**Review by dharmabumstead — APPROVED** [review]
Looks OK from a docs standpoint. Thanks @tedder!

**Comment by mattclay** [comment]
The PEP 8 tests can now be run with `make pep8` from the git root.

File: test/sanity/pep8/README.md [hunk 1]
```diff
@@ -24,3 +24,10 @@ Files listed in the [skip list](skip.txt) are not tested by `pep8`.
 ## Removed Files
 
 Files which have been removed from the repository must be removed from the legacy file list and the skip list.
+
+## Running locally
```
Comment (mattclay) [inline test/sanity/pep8/README.md]: Capitalize `locally` for consistency with other headings.

File: test/sanity/pep8/README.md [hunk 2]
```diff
@@ -24,3 +24,10 @@ Files listed in the [skip list](skip.txt) are not tested by `pep8`.
...
+
+You can do a quick pep8 spotcheck by approximating `command_sanity_pep8` in `test/runner/lib/executor.py` like this:
```
Comment (mattclay) [inline test/sanity/pep8/README.md]: No need to reference the internals of `ansible-test` here. A simpler explanation should suffice:

```
The PEP 8 check can be run locally with:
```

File: test/sanity/pep8/README.md [hunk 3]
```diff
@@ -24,3 +24,10 @@ Files listed in the [skip list](skip.txt) are not tested by `pep8`.
...
+
+    ./test/runner/ansible-test sanity -v --test pep8 <files to sanity-check>
```
Comment (mattclay) [inline test/sanity/pep8/README.md]: I'd revise the example a little. Verbose output is probably not needed in most cases:

```
test/runner/ansible-test sanity --test pep8 [file-or-directory-path-to-check] ...
```

**Comment by tedder** [comment]
@mattclay is there support for only running against one file/directory?

**Review by mattclay — CHANGES_REQUESTED** [review]

**Comment by mattclay** [comment]
@tedder For that, I recommend running `ansible-test` directly, so you can take advantage of tab completion provided by `argcomplete`. However, if you do want to use `make`, you can pass extra arguments using the `TEST_FLAGS` environment variable.

The `pep8` target was already present, I just recently updated it to use `ansible-test`.

**Comment by tedder** [comment]
Made those corrections. Originally the "approximating command_sanity_pep8" text was because.. I was running a custom command to approximate it. Yay for built-in methods to do so.

**Comment by mattclay** [comment]
@tedder Please remove the '@' mention from the commit message to avoid spurious notifications from GitHub.

**Comment by tedder** [comment]
done (and force-pushed)

File: test/sanity/pep8/README.md [hunk 4]
```diff
@@ -24,3 +24,10 @@ Files listed in the [skip list](skip.txt) are not tested by `pep8`.
 ## Removed Files
 
 Files which have been removed from the repository must be removed from the legacy file list and the skip list.
+
+## Running Locally
+
+The pep8 check can be run locally with:
+
+    ./test/runner/ansible-test sanity --test pep8 <files to sanity-check>
```
Comment (mattclay) [inline test/sanity/pep8/README.md]: The example in my previous review comment probably explains usage better:

```
test/runner/ansible-test sanity --test pep8 [file-or-directory-path-to-check] ...
```

It makes it more clear that paths given are:

- optional
- can be files or directories
- can be specified more than once

**Review by mattclay — CHANGES_REQUESTED** [review]

File: test/sanity/pep8/README.md [hunk 5]
```diff
@@ -24,3 +24,10 @@ Files listed in the [skip list](skip.txt) are not tested by `pep8`.
 ## Removed Files
 
 Files which have been removed from the repository must be removed from the legacy file list and the skip list.
+
+## Running Locally
+
+The pep8 check can be run locally with:
+
+    ./test/runner/ansible-test sanity --test pep8 [file-or-directory-path-to-check]
```
Comment (mattclay) [inline test/sanity/pep8/README.md]: Missing `...` at the end to indicate multiple files can be specified.

**Review by mattclay — CHANGES_REQUESTED** [review]

**Review by mattclay — APPROVED** [review]

**Pull request merged**